FMSgridtools/make_hgrid/hgridobj.py

```diff
@@ -0,0 +1,288 @@
+import dataclasses
+from typing import List, Optional
+import numpy as np
+import numpy.typing as npt
+import xarray as xr
+
+from FMSgridtools.shared.gridtools_utils import check_file_is_there
+from FMSgridtools.shared.gridobj import GridObj
+
```

Comment:
I might be missing something, is GridObj being used at all?  Shouldn't HGridObj be populating GridObj or use GridObj?